mocks/sa.go

```diff
@@ -271,6 +271,16 @@ func (sa *StorageAuthority) GetRevokedCerts(ctx context.Context, _ *sapb.GetRevo
 	return &ServerStreamClient[corepb.CRLEntry]{}, nil
 }
 
+// GetRevokedCertsByShard is a mock
+func (sa *StorageAuthorityReadOnly) GetRevokedCertsByShard(ctx context.Context, _ *sapb.GetRevokedCertsByShardRequest, _ ...grpc.CallOption) (sapb.StorageAuthorityReadOnly_GetRevokedCertsClient, error) {
```

Comment:
Thanks to my upstream grpc stream generics work, you shouldn't need to create two different versions of this method that differ only in their receiver and return type! Instead a single mock with [this signature from the autogenerated code](https://github.com/letsencrypt/boulder/pull/7946/files#diff-5763d37fc1c61c7afe8d717f7c511c1c3250fa43ed8b66ee46e39e0a685b4203R75) should work:
```suggestion
func (sa *StorageAuthorityReadOnly) GetRevokedCertsByShard(ctx context.Context, _ *sapb.GetRevokedCertsByShardRequest, _ ...grpc.CallOption) (grpc.ServerStreamingClient[corepb.CRLEntry], error) {
```
And once you're using that generic return type, both methods can be collapsed into a single method with a `StorageAuthorityReadOnly` receiver because the mock `StorageAuthority` embeds the former.

Reply:
@jsha I'm holding off on merging this so that you can consider whether you would like to address this comment.